runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/validator/StringValidator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021-2023 Aklivity Inc.
+ *
+ * Aklivity licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.aklivity.zilla.runtime.binding.kafka.internal.validator;
+
+import org.agrona.DirectBuffer;
+
+import io.aklivity.zilla.runtime.binding.kafka.internal.validator.config.StringValidatorConfig;
+
+public final class StringValidator implements Validator
+{
+    private String encoding;
+
+    public StringValidator(
+        StringValidatorConfig config)
+    {
+        this.encoding = config.encoding;
```

Comment:
Perhaps change field type to `Predicate<byte[]>` and select `this::isValidUTF8` for `utf-8` with a default implementation for other charsets.

Then `validate` method would be something like
```
byte[] array = new byte[length];
data.getBytes(0, array);
return predicate.test(array);
```